frontend/userpatch.lua

```diff
@@ -0,0 +1,138 @@
+        early = "1",       -- to be started early on startup (always, but after an `early_once`)
+        late = "2",        -- to be started after UIManager is ready (always)
+                           -- 3-7 are reserved for later use
+        before_exit = "8", -- to be started a bit before exit before settings are saved (always)
+        on_exit = "9",     -- to be started right before exit (always)
+
+        applyPatches = function(priority) end, -- to be overwritten, if the device allows it.
+    }
+
+if isAndroid and android.prop.flavor == "fdroid" then
+    return userpatch -- allows to use applyPatches as a no-op on F-Droid flavor
+end
+
+------------------------------------------------------------------------------------
+
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local DataStorage = require("datastorage")
+
+-- the directory KOReader is installed in (and runs from)
+local package_dir = lfs.currentdir()
+-- the directory where KOReader stores user data (on SDL we may set `XDG_DOCUMENTS_DIR=~/koreader/`)
+local data_dir = os.getenv("XDG_DOCUMENTS_DIR") or DataStorage:getDataDir() or package_dir
+
+-- A wrapper for os.execute, can not use ffi/util early before setupkoenv.
+local function execute(...)
+    local command = {}
+    for i, v in ipairs({...}) do
+        table.insert(command, "'")
+        table.insert(command, tostring(v))
+        table.insert(command, "' ")
+    end
+    local cmd = table.concat(command)
+    logger.info("Live update apply:", cmd)
+    local retval =  os.execute(cmd)
+    logger.info("Live update script returned:", retval)
+end
+
+--- Run user shell scripts or lua patches
+-- Execution order order is alphanum-sort for humans version 4: `1-patch.lua` is executed before `10-patch.lua`
+-- (see http://notebook.kulchenko.com/algorithms/alphanumeric-natural-sorting-for-humans-in-lua)
+-- string directory ... to scan through (flat no recursion)
+-- string priority ... only files starting with `priority` followed by digits and a '-' will be processed.
+-- bool update_once_pending ... if true the onetime update has not been executed
+-- string update_once_marker ... remove this file, if onetime update gets applied the first time
+local function runLiveUpdateTasks(dir, priority, update_once_pending, update_once_marker)
+    if lfs.attributes(dir, "mode") ~= "directory" then
+        return
+    end
+
+    local patches = {}
+    for entry in lfs.dir(dir) do
+        local mode = lfs.attributes(dir .. "/" .. entry, "mode")
+        if entry and mode == "file" and entry:match("^" .. priority .. "%d*%-") then
+            table.insert(patches, entry)
+        end
+    end
+
+    if #patches == 0 then
+        return -- nothing to do
+    end
+
+    local function addLeadingZeroes(d)
+        local dec, n = string.match(d, "(%.?)0*(.+)")
+        return #dec > 0 and ("%.12f"):format(d) or ("%s%03d%s"):format(dec, #n, n)
+    end
+    local sorting = function(a, b)
```

Comment:
Make these 2 local functions have the same look :)
(Either `local function z()` or `local z = function()`, your choice, unless there is a favored one in our code.)